packages/store/src/indexer/search.ts

```diff
@@ -112,17 +112,19 @@ export class SearchIndexer {
     if (!(yBlocks instanceof YMap)) {
       return;
     }
-    yBlocks.forEach((_, key) => {
-      this._refreshIndex(pageId, key, 'add', yBlocks.get(key));
-    });
+    if (yBlocks instanceof YMap) {
+      yBlocks.forEach((_, key) => {
+        this._refreshIndex(pageId, key, 'add', yBlocks.get(key));
+      });
+    }
   }
 
   private _handlePageIndexing(pageId: string, page?: Doc) {
     if (!page) {
       return;
     }
     const yBlocks = page.get('blocks');
```

Comment:
@himself65  You can simply do `page.get('blocks', Y.Map)` and always retrieve a Y.Map (even if another user uses this as a `Y.Text`).

Reply:
See https://github.com/toeverything/blocksuite/pull/4296